app/controllers/service_types_controller.rb

```diff
@@ -3,8 +3,40 @@ class ServiceTypesController < ApplicationController
 
   # Authentication by default inherited from ApplicationController.
 
+  before_action :authorize_for_districtwide_access_admin, except: [:index]
+
+  def authorize_for_districtwide_access_admin
+    unless current_educator.admin? && current_educator.districtwide_access?
+      error = 'You don\'t have the correct authorization.'
+
+      return render json: { error: error }, status: 403
+    end
+  end
+
   def index
     render json: ServiceType.pluck(:name).sort
   end
 
+  def is_service_working_json
+    service_type_id = params[:service_type_id]
+
+    service_type = ServiceType.find(service_type_id)
+    student_ids = service_type.services.where('date_started > ?', Time.current - 1.year).map(&:student_id)
+
+    chart_data = Student.active.where(id: student_ids).map do |student|
```

Comment:
This is another "unscoped" query, so good to wrap it with `authorized`.  But this seems like it could collapse with the one up above into a single query:

```
students = authorized do
  service_type.services.where('date_started > ?', Time.current - 1.year).map(&:students).select(&:active?)
end
```